ninja-core/src/main/java/ninja/validation/Validator.java

```diff
@@ -32,9 +34,24 @@
      * @param validation
      *            The validation context
      */
+    @Deprecated
     void validate(T value, String field, Validation validation);
 
     /**
+     * Validate the given value
+     *
+     * @param value
+     *            The value, may be null
+     * @param field
+     *            The name of the field being validated, if applicable
+     * @param validation
+     *            The validation context
+     * @param context
+     *            The Ninja request context
+     */
+    void validate(T value, String field, Validation validation, Context context);
```

Comment:
Having validation AND context seems redundant - could we just have context (which got context.getValidation?).

That's of course not really backwards compatible - do you think we can make that? I actually don't have big problems with breaking backwards compatibility in a - say Ninja 4.1 release.